cmd/config.go

```diff
@@ -274,3 +275,97 @@ func validateScenarioConfig(conf lib.ExecutorConfig, isExecutable func(string) b
 	}
 	return nil
 }
+
+func validateThresholdsConfig(conf Config, registry *metrics.Registry) error {
+	// Registry should not be nil under any circumstances.
+	// N.B: the reason for passing it as a pointer in the first place is
+	// because it holds a Mutex, which effectively forbids passing it by value.
+	if registry == nil {
+		err := fmt.Errorf(
+			"unable to validate thresholds configuration; reason: " +
+				"provided registry is nil. It looks like you have found a bug, " +
+				"please open an issue on our GitHub: " +
+				"https://github.com/grafana/k6/issues/new?assignees=&labels=bug&template=bug.yaml",
+		)
+		return errext.WithExitCodeIfNone(err, exitcodes.InvalidConfig)
+	}
+
+	// Focusing on builtin metric for now. For each threshold,
+	// found in the options, do we have a matching builtin metric?
+	for name, thresholds := range conf.Thresholds {
+		// The defined threshold applies to a non-existing metrics
+		if !registry.Has(name) {
+			err := fmt.Errorf("invalid threshold defined on %s; reason: no metric named %s found", name, name)
+			return errext.WithExitCodeIfNone(err, exitcodes.InvalidConfig)
+		}
+
+		// Get the metric from the registry
+		metric := registry.Get(name)
+
+		switch metric.Type {
```

Comment:
I would take any idea on how to improve here. I wanted to return thorough, exhaustive, user-friendly error messages, but it cascaded quickly into LONG multiline statements. I've had ideas, but none really satisfying: feedback welcome 🙏🏻